src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1196,12 +1195,31 @@ private void CheckboxPrerelease_CheckChanged(object sender, EventArgs e)
             RegistrySettingUtility.SetBooleanSetting(Constants.IncludePrereleaseRegistryName, _topPanel.CheckboxPrerelease.IsChecked == true);
             NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
-                await SearchPackagesAndRefreshUpdateCountAsync(useCacheForUpdates: false);
-                sw.Stop();
-                EmitRefreshEvent(timeSpan, RefreshOperationSource.CheckboxPrereleaseChanged, RefreshOperationStatus.Success, isUIFiltering: false, sw.Elapsed.TotalMilliseconds);
+                await RunAndEmitRefreshAsync(async () => await SearchPackagesAndRefreshUpdateCountAsync(useCacheForUpdates: false),
+                    RefreshOperationSource.CheckboxPrereleaseChanged, timeSpan, sw);
             }).PostOnFailure(nameof(PackageManagerControl), nameof(CheckboxPrerelease_CheckChanged));
         }
 
+        private async Task RunAndEmitRefreshAsync(Func<Task> runner, RefreshOperationSource source, TimeSpan lastRefresh, Stopwatch sw, bool isUIFiltering = false)
+        {
+            var refreshStatus = RefreshOperationStatus.NoOp;
+            try
+            {
+                await runner();
+                refreshStatus = RefreshOperationStatus.Success;
+            }
+            catch
+            {
+                refreshStatus = RefreshOperationStatus.Failed;
```

Comment:
Nice! 

I'm a little curious how this Failure can happen given that we swallow exceptions here: https://github.com/NuGet/NuGet.Client/blob/a1e6c481151d686d507ddfbaa1d5b15658582bd8/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L887

Maybe this should be a more generic Telemetry utility?

Reply:
I imagine those swallow cancellationToken exceptions. This method was inspired by scenario: When Solution PM UI is opened, visible and a project is renamed. I noticed this scenario is often marked as a failure, so, I created this method and now it logs the operation as a failure

Reply:
Oh you're right, thought it was all Exceptions at first glance.